### PR TITLE
Promotion from the UI

### PR DESCRIPTION
**Pull request by kirillta**
The very basic implementation of a promotion feature

image

image

### CODE REVIEW

**Comment by bradrydzewski** [comment]
thanks! I will review the pull request and provide feedback. One thing I noticed is that the `package-lock.json` file was deleted and should be restored.

**Comment by kirillta** [comment]
I can restore it, but the project doesn't builds with the current `package-lock.json`

**Comment by kirillta** [comment]
@bradrydzewski may I have your attention here?

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,5 @@ dist/files/*
 *.log
 .env
 .env.*
-hermione
\ No newline at end of file
+hermione
+.vscode/launch.json
```
Comment (tboerger) [inline .gitignore:8]: Editor specific ignores should be written to your personal global gitignore.

File: src/components/editable-list/EditableListItemWithValue.vue [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<!-- A list item of editable list -->
+
+<template>
+  <div class="editable-list-item">
+    <span class="name">{{ name }}</span>
+
+    <span class="name">{{ value }}</span>
```
Comment (tboerger) [inline src/components/editable-list/EditableListItemWithValue.vue:7]: Class name for the value?

**Comment by kirillta** [comment]
@tboerger done

**Comment by kirillta** [comment]
image
I added build variables support

**Comment by kowsheek** [comment]
So excited for this! Thank you @kirillta @tboerger @bradrydzewski 🙏

**Comment by brandoncollins7** [comment]
How does one build and deploy this version of the ui? I can't seem to figure out how to build the .go files:(

**Comment by kirillta** [comment]
> How does one build and deploy this version of the ui? I can't seem to figure out how to build the .go files:(

I'm sorry, repository owners ignore the PR and it became outdated. As I understand they made their own version of the deployment screen. It will be available when you'll update Drone UI

**Comment by ashwilliams1** [comment]
@kirillta we were able to use pull request #329 and simply removed some autocomplete features that we did not want included in the initial release. Support for the deployment tab and deployment popup are available in master.

**Pull request closed without merging**